hydroflow/src/scheduled/handoff/mod.rs

```diff
@@ -18,7 +16,9 @@ pub trait CanReceive<T> {
 /**
  * A handle onto the metadata part of a [Handoff], with no element type.
  */
-pub trait HandoffMeta {
+pub trait HandoffMeta: Any {
+    fn any_ref(&self) -> &dyn Any;
```

Comment:
How come this can't be done at the call site?

Reply:
It is weird but necessary unfortunately. Basically we have the concrete type `dyn HandoffMeta` but  [`downcast_ref`](https://doc.rust-lang.org/std/any/trait.Any.html#implementations) is on `dyn Any`. If we call it directly we get the impl on `dyn HandoffMeta`, so we either need [trait upcasting (unstable)](https://doc.rust-lang.org/nightly/unstable-book/language-features/trait-upcasting.html) or just have to manually implement this indirection. Longer version here: https://stackoverflow.com/a/33687996/2398020